fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java

```diff
@@ -317,12 +319,43 @@ public static ConstantOperator dateParse(ConstantOperator date, ConstantOperator
                 // If parsing fails, it can be re-parsed from the position of the successful prefix string.
                 // This way datetime string can use incomplete format
                 // eg. str_to_date('2022-10-18 00:00:00','%Y-%m-%d %H:%s');
-                ldt = LocalDateTime.from(builder.withResolverStyle(ResolverStyle.STRICT)
-                        .parse(dateStr.substring(0, e.getErrorIndex())));
+                if (dateStr.length() > e.getErrorIndex()) {
+                    ldt = LocalDateTime.from(builder.withResolverStyle(ResolverStyle.STRICT)
+                            .parse(dateStr.substring(0, e.getErrorIndex())));
+                } else {   // eg. str_to_date('2022-10-18 00:00','%Y-%m-%d %H:%s:%i');
+                    int length = dateStr.length();
+                    //Since the length of the actual format cannot be obtained, it needs to be filled first. If it fails again,
+                    //the length of the actual format can be obtained, and the format can be converted.
+                    try {
+                        ldt = LocalDateTime.from(builder.withResolverStyle(ResolverStyle.STRICT)
+                                .parse(dateStr + DEFAULT_DATE_TIME.substring(length)));
+                    } catch (DateTimeParseException pe) {
+                        ldt = LocalDateTime.from(builder.withResolverStyle(ResolverStyle.STRICT)
+                                .parse(dateStr + DEFAULT_DATE_TIME.substring(length, pe.getErrorIndex())));
+                    }
+                }
             }
             return ConstantOperator.createDatetime(ldt, Type.DATETIME);
         } else {
-            LocalDate ld = LocalDate.from(builder.withResolverStyle(ResolverStyle.STRICT).parse(dateStr));
+            LocalDate ld;
+            try {
+                ld = LocalDate.from(builder.withResolverStyle(ResolverStyle.STRICT).parse(dateStr));
+            } catch (DateTimeParseException e) {
+                // eg. str_to_date('2022-10-18 00:00:00','%Y-%m-%d');
+                if (dateStr.length() > e.getErrorIndex()) {
+                    ld = LocalDate.from(builder.withResolverStyle(ResolverStyle.STRICT)
+                            .parse(dateStr.substring(0, e.getErrorIndex())));
+                } else {   // eg. str_to_date('2022-10','%Y-%m-%d');
```

Comment:
`str_to_date('2022-10','%Y-%m-%d')` returns `null` in BE, so we needn't this `else` branch.

```suggestion
                ld = LocalDate.from(builder.withResolverStyle(ResolverStyle.STRICT)
                            .parse(dateStr.substring(0, e.getErrorIndex())));
```

Reply:
You can use `set disable_function_fold_constants=off/on` to check whether FE/BE's behavior is the same or not.